src/_data/cms.js

```diff
@@ -54,23 +54,41 @@ async function _fetchDevotionRangeFromUrl(url, devotions) {
   return devotions;
 }
 
+function calculateDateRange(currentDate) {
+  const startDayOffset = 4;
+  const endDayOffset = 3;
+
+  const start = new Date(currentDate.getFullYear(), 0, 0);
+  const diff = currentDate - start;
+  const dayOfYear = Math.floor(diff / (1000 * 60 * 60 * 24));
+
+  let startDay, endDay;
+
+  if (dayOfYear <= 5) {
+    startDay = 1;
```

Comment:
How about avoiding all those else statements with early returns like this:

```suggestion
    return { 1, 8 };
```

Reply:
Nice suggestion.